### PR TITLE
New submission flow: Your test available screen (EXPOSUREAPP-3741)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
Your test available screen.

Your_test_result_available

### CODE REVIEW

**Review by Oliver-Zimmerman — DISMISSED** [review]
LGTM 👍

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/resultavailable/SubmissionTestResultAvailableFragment.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+class SubmissionTestResultAvailableFragment : Fragment(R.layout.fragment_submission_test_result_available), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val vm: SubmissionTestResultAvailableViewModel by cwaViewModels { viewModelFactory }
+    private val binding: FragmentSubmissionTestResultAvailableBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        vm.consent.observe2(this) {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/resultavailable/SubmissionTestResultAvailableFragment.kt:26]: For better readability, I'd propose to either rename `vm.consent` to `vm.consentGiven` or give `it` a proper name

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/resultavailable/SubmissionTestResultAvailableFragment.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            binding.submissionTestResultAvailableConsentStatus.consent = it
+        }
+
+        binding.apply {
+            submissionTestResultAvailableProceedButton.setOnClickListener { vm.proceed() }
+            submissionTestResultAvailableHeader.headerButtonBack.buttonIcon.setOnClickListener { vm.goBack() }
+            submissionTestResultAvailableConsentStatus.setOnClickListener { vm.goConsent() }
+        }
+
+        vm.clickEvent.observe2(this) {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/resultavailable/SubmissionTestResultAvailableFragment.kt:43]: technically, it is a navigation event, not a click event, isn't it?

File: Corona-Warn-App/src/main/res/layout/fragment_submission_test_result_available.xml [hunk 3]
```diff
@@ -0,0 +1,106 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/res/layout/fragment_submission_test_result_available.xml:2]: Why do we need  a layout around the constraint layout?

**Review by ralfgehrer — APPROVED** [review]
Screen is implemented according to design.
Minor comments.

**Review by Oliver-Zimmerman — APPROVED** [review]

**Pull request merged**